enhancements/builds/remove-jenkins-payload.md

```diff
@@ -0,0 +1,239 @@
+
+Once the CPaaS pipeline is ready, the Jenkins templates and imagestreams delivered in OpenShift via Samples operator will need to reference the container catalog images (just like other layered products).
+The Samples Operator can then remove its special logic for Jenkins imagestreams and templates.
+
+#### Jenkins / OCP Compatibility
+
+The label `io.openshift.versions` will be added to the Jenkins image via a Dockerfile LABEL instruction.
+Its value will be an inclusive range of OCP major+minor versions that the image is officially supported on.
+For example, the current Jenkins image would set the value `v4.6-v4.8`.
+
+In the event a breaking change is introduced, the `ocp-*` backport branch will maintain the Dockerfile that sets the appropriate OCP supported version label.
+The main branch will then update the `io.openshift.versions` label to indicate that prior OCP versions are no longer supported.
+For example, if a breaking change is introduced in OCP 4.9, the branch `ocp-4.8` will be added to all Jenkins repositories.
+The main branch will then be update the `io.openshift.versions` label on the image to `v4.9`, while the image produced from the `ocp-4.8` branch will continue to use `v4.6-v4.8`.
+
+#### Jenkins Plugin Version Scheme
+
+Jenkins plugins for OpenShift currently use the 1.0.<z> versioning scheme.
+This versioning scheme will be updated so that the minor Y version corresponds to the lowest supported version of OCP 4 for the plugin.
+For example, the next release of the openshift-client plugin should be 1.6.0, since OCP 4.6 is the lowest supported version of OCP for the plugin (that we can support).
+
+The main/master branch for Jenkins and its plugins will be assumed to support the version of OpenShift under development.
+If a breaking change is introduced, a separate branch named `ocp-4.n` should be created to handle bug backports, with `n` representing the highest supported version of OpenShift.
+The requisite delivery pipelines will also need to be updated so that bugfix builds can be issued from `ocp-*` branches.
+
+#### Jenkins Image Tagging Scheme
+
+The Jenkins image will use the supported OCP version as the basis of its tagging scheme:
+
+1. `latest` - latest image build
+2. `4` - latest image that supports OCP 4
+3. `4.<Y>`- latest image that supports up to OCP 4.Y
+4. `4.<Y>.0-YYYYMMDDHHMM-<git-sha>` - latest image that supports up to OCP 4.Y, produced at the provided datetime YYYYMMDDHHMM format and source commit sha.
+
+CPaaS and Errata will be configured so that tags for `4.Y` are set appropriately for the given branch, based on the `io.openshift.versions` Dockerfile label above.
+For example, if the Dockerfile has the `v4.6-v4.9` version label, the image produced by CPaaS will have the following tags:
+
+- `latest`
+- `4`
+- `4.6`, `4.7`, `4.8`, and `4.9`
+- `4.6.0-YYYYMMDDHHMM-<git-sha>`, `4.7.0-YYYYMMDDHHMM-<git-sha>`, `4.8.0-YYYYMMDDHHMM-<git-sha>`, `4.9.0-YYYYMMDDHHMM-<git-sha>`
```

Comment:
@akram please verify that this tagging scheme if feasible. If yes, then you can go ahead and add lgtm to this proposal.

Reply:
@adambkaplan I don't see CPaaS being able to specify a custom tagging scheme. It relies on OSBS/brew at the end of the day, and we just got tags like:
`jenkins-4.8-ordinal` where ordinal is an incremented sequence from and by brew.

Reply:
Updated so the tag scheme uses the ordinal from brew